macos/FluentUI/Badge/BadgeView.swift

```diff
@@ -0,0 +1,154 @@
+	///   - style: The BadgeViewStyle
+	@objc public init(title: String, style: Style) {
+		textField = NSTextField(labelWithString: title)
+		switch style {
+		case .default:
+			_backgroundColor = BadgeColors.defaultBackground
+			_textColor = BadgeColors.defaultText
+		case .primary:
+			_backgroundColor = BadgeColors.primaryBackground
+			_textColor = BadgeColors.primaryText
+		}
+		super.init(frame: .zero)
+
+		wantsLayer = true
+		layer?.cornerRadius = Size.small.cornerRadius
+
+		textField.lineBreakMode = .byTruncatingMiddle
+		textField.alignment = .center
+		textField.backgroundColor = .clear
+		textField.translatesAutoresizingMaskIntoConstraints = false
+		textField.font = NSFont.systemFont(ofSize: Size.small.fontSize, weight: .light)
+		addSubview(textField)
+
+		setAccessibilityElement(true)
+		setAccessibilityLabel(title)
+		setAccessibilityRole(.staticText)
+
+		var horizontalPadding: CGFloat
+		var verticalPadding: CGFloat
+		horizontalPadding = Size.small.horizontalPadding
+		verticalPadding = Size.small.verticalPadding
+		NSLayoutConstraint.activate([
+			textField.leadingAnchor.constraint(equalTo: leadingAnchor, constant: horizontalPadding),
+			textField.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -horizontalPadding),
+			textField.topAnchor.constraint(equalTo: topAnchor, constant: verticalPadding),
+			textField.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -verticalPadding)
+		])
+	}
+
+	public required init?(coder aDecoder: NSCoder) {
+		preconditionFailure("init(coder:) has not been implemented")
+	}
+
+	open override func updateLayer() {
+		updateColors()
+	}
+
+	private var _backgroundColor: DynamicColor
+	@objc open var backgroundColor: DynamicColor {
+		get {
+			return _backgroundColor
+		}
+		set {
+			if backgroundColor != newValue {
+				_backgroundColor = newValue
+				updateColors()
+			}
+		}
+	}
+
+	private var _textColor: DynamicColor
+	@objc open var textColor: DynamicColor {
+		get {
+			return _textColor
+		}
+		set {
+			if textColor != newValue {
+				_textColor = newValue
+				updateColors()
+			}
+		}
+	}
+
+	private var textField: NSTextField
+
+	private func updateColors() {
+		updateBackgroundColors()
+		updateTextColors()
+	}
+
+	private func updateBackgroundColors() {
+		layer?.backgroundColor = backgroundColor.resolvedColor(window?.effectiveAppearance).cgColor
+	}
+
+	private func updateTextColors() {
+		textField.textColor = textColor.resolvedColor(window?.effectiveAppearance)
+	}
+
+	private enum Size: Int, CaseIterable {
+		case small
+
+		var fontSize: CGFloat {
+			switch self {
+			case .small:
+				return 11
+			}
+		}
+
+		var cornerRadius: CGFloat {
+			switch self {
+			case .small:
+				return 3
+			}
+		}
+
+		var verticalPadding: CGFloat {
+			switch self {
+			case .small:
+				return 3
+			}
+		}
+
+		var horizontalPadding: CGFloat {
+			switch self {
+			case .small:
+				return 4
+			}
+		}
+	}
+}
+
+// MARK: - Colors
+
+struct BadgeColors {
```

Comment:
might as well explictly set it private so only this file can access.

Reply:
Or maybe even moving the struct definition to the BadgeView class and mark it private.
Maybe rename it to "Constants"?